### PR TITLE
Make the model read-only after Finalize

### PR DESCRIPTION
**Pull request by AndriySvyryd**
Part of #8258

### CODE REVIEW

File: test/EFCore.Tests/ModelBuilding/NonRelationshipTestBase.cs [hunk 1]
```diff
@@ -33,6 +33,16 @@ public void Can_set_model_annotation()
...
             }
 
+            [Fact]
+            public void Model_is_readonly_after_Finalize()
+            {
+                var modelBuilder = CreateModelBuilder();
+
+                modelBuilder.Finalize();
+
+                Assert.ThrowsAny<Exception>(() => modelBuilder.HasAnnotation("Fus", "Ro"));
```
Comment (ajcvickers) [inline test/EFCore.Tests/ModelBuilding/NonRelationshipTestBase.cs]: What does this actually throw and how?

Comment (AndriySvyryd) [reply test/EFCore.Tests/ModelBuilding/NonRelationshipTestBase.cs]: It throws `NullReferenceException` because the `ConventionDispatcher` is null.

Comment (ajcvickers) [reply test/EFCore.Tests/ModelBuilding/NonRelationshipTestBase.cs]: This is presumably a temporary thing?

Comment (AndriySvyryd) [reply test/EFCore.Tests/ModelBuilding/NonRelationshipTestBase.cs]: Yes, #8258 will add a proper message when implemented (post 3.0). This just avoids the future breaking change.

**Pull request merged**